lib/core/class-affiliates-shortcodes.php

```diff
@@ -711,21 +720,24 @@ public static function affiliates_url( $atts, $content = null ) {
 				intval( $user_id )
 			))) {
 				if ( strlen( $content ) == 0 ) {
-					$base_url = get_bloginfo( 'url' );
+					$url = get_bloginfo( 'url' );
+					if ( $current !== null ) {
+						$url = get_permalink();
```

Comment:
@eggemplo Using get_permalink() won't bring the result we would expect in several cases as far as I can remember. We have several instances of obtaining the URL in a more robust manner, `$current_url` in existing code